File: Baekjoon_Python/Baekjoon_Python/1322.py
X, K = map(int, input().split())

def solution(X, K):
    s = ''

    while K:
        if(X & 1):
            s = '0' + s
        else:
            s = str(K&1) + s
            K >>= 1
        X >>= 1
    
    return int(s, base=2)

print(solution(X, K))

File: Baekjoon_Javascript/1322.js
const file = require('fs').readFileSync('/dev/stdin');
const input = file.toString().trim().split(' ');

const [X, K] = input.map(v=>BigInt(v));

const solution = (X, K) => {
    let str = '';

    while(K) {
        if(X & 1n) str = '0' + str;
        else {
            str = (K & 1n) + str;
            K >>= 1n;
        }
        X >>= 1n;
    }

    return `${BigInt('0b'+str)}`;
}

console.log(solution(X, K));